cpp/include/cudf/io/datasource.hpp

```diff
@@ -36,6 +37,212 @@ namespace io {
  * @file
  */
 
+/**
+ * @brief Kind of data source to create when calling
+ * `cudf::io::datasource::create()`.
+ *
+ * @see cudf::io::datasource::create()
+ * @see cudf::io::datasource_params
+ *
+ * N.B. GDS = GPUDirect Storage
+ */
+enum class datasource_kind {
```

Comment:
I notice that an equivalent of what is currently env var "GDS" is mising :)
I think that's fine, just wanted to leave a not for viz.

Reply:
Ah yeah that's basically just `cuFile` device I/O for everything right?  I think that's what I was referring to above when I said it would make sense to probably have a `CUFILE` `datasource_kind` as well.

Reply:
Just added `datasource_kind::GDS` and corresponding (very simple) `cufile_source` implementation.